ParProcCo/msmapper_aggregator.py

```diff
@@ -54,18 +57,61 @@ def _check_total_slices(self, total_slices: int, output_data_files: List[Path])
     def _renormalise(self, output_data_files: List[Path]) -> None:
         self.output_data_files = output_data_files
         self._get_nxdata()
-        self._fill_axes_fields()
-        self._initialise_accumulator_arrays()
+        self._initialise_arrays()
         self._accumulate_volumes()
 
+    def _initialise_arrays(self) -> None:
+        self.axes_mins = [np.inf] * self.data_dimensions
+        self.axes_maxs = [np.NINF] * self.data_dimensions
+        self._axes_starts_stops = [[] for _ in range(self.data_dimensions)]
+        self.all_slices = []
+
+        for data_file in self.output_data_files:
+            with h5py.File(data_file, "r") as f:
+                data_group_name = "/".join([self.nxentry_name, self.nxdata_name])
+                nx_data_group = f[data_group_name]
+                volumes_array = np.array(nx_data_group[self.signal_name])
+                axes = [np.array(nx_data_group[axis_name]) for axis_name in self.axes_names]
+
+            axes_lengths = volumes_array.shape
+            slices = []
+
+            for j, axis in enumerate(axes):
+                self.axes_mins[j] = min([min(axis), self.axes_mins[j]])
+                self.axes_maxs[j] = max([max(axis), self.axes_maxs[j]])
+                self._axes_starts_stops[j].append([axis[0], axis[-1]])
+
+                start = int((axis[0] - self.axes_mins[j]) / self.axes_spacing[j])
+                stop = axes_lengths[j] - start
+                slices.append(slice(start, stop))
+
+            assert len(slices) == self.data_dimensions
+            self.all_slices.append(slices)
+
+        self.accumulator_axis_lengths = []
+        self.accumulator_axis_ranges = []
+
+        for i in range(self.data_dimensions):
+            length = int((self.axes_maxs[i] - self.axes_mins[i]) / self.axes_spacing[i]) + 1
+            self.accumulator_axis_lengths.append(length)
+            ranges = [(round((x * self.axes_spacing[i]), 4) + self.axes_mins[i])
+                      for x in range(self.accumulator_axis_lengths[i])]
+            self.accumulator_axis_ranges.append(ranges)
+
+        for j, axis in enumerate(axes):
+            if not np.allclose(axis, self.accumulator_axis_ranges[j][slices[j]]):
+                raise RuntimeError
```

Comment:
This check should be per-file axes. Also want more information in error message here